module/zfs/dmu_zfetch.c

```diff
@@ -312,7 +329,29 @@ dmu_zfetch(zfetch_t *zf, uint64_t blkid, uint64_t nblks, boolean_t fetch_data)
 		 */
 		pf_ahead_blks = zs->zs_pf_blkid - blkid + nblks;
 		max_blks = max_dist_blks - (pf_start - end_of_access_blkid);
+
+		if (dmu_zfetch_out_of_range == 0 && max_blks < 0) {
+			toss4454_warn("data:max_blks < 0", blkid, nblks,
+			    orig_zs_blkid, orig_zs_pf_blkid, orig_zs_ipf_blkid,
+			    max_dist_blks);
+			dmu_zfetch_out_of_range = 1;
```

Comment:
`dmu_zfetch_out_of_range = 1;` gets set within `toss4454_warn()`.  Did you want to set it in both places?

Reply:
Nope, fixed.